tests/estimator/ops/test_estimator_parametric_single_qubit.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for parametric single qubit resource operators.
+"""
+
+import pytest
+
+from pennylane.estimator.ops import GlobalPhase, Hadamard, Identity, T
+from pennylane.estimator.ops.qubit.parametric_ops_single_qubit import (
+    RX,
+    RY,
+    RZ,
+    PhaseShift,
+    Rot,
+    _rotation_resources,
+)
+from pennylane.estimator.resource_operator import CompressedResourceOp, GateCount
+from pennylane.exceptions import ResourcesUndefinedError
+
+# pylint: disable=no-self-use, use-implicit-booleaness-not-comparison,too-many-arguments
+
+params = list(zip([10e-3, 10e-4, 10e-5], [17, 21, 24]))
+
+
+@pytest.mark.parametrize("precision, expected", params)
+def test_rotation_resources(precision, expected):
+    """Test the hardcoded resources used for RX, RY, RZ"""
+    gate_types = [GateCount(CompressedResourceOp(T, 1, {}), expected)]
+
+    assert gate_types == _rotation_resources(precision=precision)
+
+
+class TestPauliRotation:
+    """Test Resource operators for RX, RY, and RZ"""
+
+    params_classes = [RX, RY, RZ]
+    params_errors = [10e-3, 10e-4, 10e-5]
+    params_ctrl_res = [
+        [
+            GateCount(CompressedResourceOp(Hadamard, 1, {}), 2),
+            GateCount(CompressedResourceOp(RZ, 1, {}), 2),
+        ],
+        [
+            GateCount(CompressedResourceOp(RY, 1, {}), 2),
+        ],
+        [
+            GateCount(CompressedResourceOp(RZ, 1, {}), 2),
+        ],
+    ]
+
+    @pytest.mark.parametrize("resource_class", params_classes)
+    @pytest.mark.parametrize("precision", params_errors)
+    def test_resources(self, resource_class, precision):
+        """Test the resources method"""
```

Comment:
Mind updating all relevant test docstrings like this?
```suggestion
        """Test the resources method matches the expected number of rotation resources"""
```